skbeam/core/image.py

```
@@ -170,3 +171,121 @@ def construct_circ_avg_image(radii, intensities, dims=None, center=None,
+
+        Note
+        ----
+        This function uses a simple linear interpolation from scipy:
+            `scipy.interpolate.RegularGridInterpolator`
+            More complex interpolation techniques (i.e. splines) cannot be used
+            with this algorithm.
+
+        Returns
+        -------
+        new_img : 2d np.ndarray
+            The reconstructed image. Masked regions are filled with np.nan
+
+    '''
+    if mask is not None:
+        if mask.shape != image.shape:
+            if mask.ndim == 2:
+                raise ValueError("Mask shape ({}, {}) ".format(*mask.shape) +
+                                 "does not match expected" +
+                                 " shape of ({},{})".format(*shape))
+            else:
+                raise ValueError("Mask not right dimensions."
+                                 "Expected 2 got {}".format(mask.ndim))
+        image[np.where(mask == 0)] = np.nan
+    if shape is None:
+        if center is not None:
+            raise ValueError("Specifying a shape but not a center does not "
+                             "make sense and may lead to unexpected results.")
+        # round up, also take into account pixel size change
+        maxr_y, maxr_x = (int(np.max(radii/pixel_size[0])+.5),
+                          int(np.max(radii/pixel_size[1])+.5))
+        shape = 2*maxr_y+1, 2*maxr_x+1
+
+    if center is None:
+        center = (shape[0]-1)/2., (shape[1] - 1)/2.
+
+    # 1 . There are a few cases to consider here for the angles:
+    #   i. [0,..., 2*pi] -> [0,..,0]
+    #   ii. [0, ... , 2*pi, ...5*pi] -> [0, ..., 0, ...]
+    #   iii. [0,..., a < 2*pi] -> [0, ..., a< 2*pi] (easy)
+    #   iv. [.12, ..., 2*pi, ...] -> [0, ...]
+
+    # 1.a : subtract minimum
+    anglemin = angles[0]
+    angles -= anglemin
+    # 1.b : modulo 2*pi
+    angles = angles % (2*np.pi)
+    # 1.c : find any extra cross-overs and ignore them
+    adiff = np.where(np.diff(angles) < 0)[0]
+    if len(adiff) > 0:
```

I would just raise here, silently dropping data is no good.